src/IO/WriteBufferFromHTTPServerResponse.h

```diff
@@ -70,6 +70,7 @@ class WriteBufferFromHTTPServerResponse final : public BufferWithOwnMemory<Write
     Progress accumulated_progress;
     size_t send_progress_interval_ms = 100;
     Stopwatch progress_watch;
+    Stopwatch total_watch;
```

Comment:
Maybe it's better to put into `Context`, set [here](https://github.com/ClickHouse/ClickHouse/blob/94c733bac76c2a33b1d4a79cafdfcfdad6f1bf10/src/Server/HTTPHandler.cpp#L735) and propagate in callback. But we can loose piece of time interval in summary if callback wouldn't called in time.

Reply:
Yeah, that makes sense. Which callback are you referring to?

Reply:
Meanwhile, it looks like the instance of `WriteBufferFromHTTPServerResponse` is created before the query execution start, so the stopwatch start time should be adequate...